### PR TITLE
Fix incorrect time exceedance check

### LINKED ISSUE #39: Missing Pred_, Succ_Transfer and vacation in time exceedance check

### PR DESCRIPTION
**Pull request by gstuer**
As stated out by #39 the check for total time exceedance or monthly time exceedance respectively, does not take vacation, predecessor- and successor-transfer into account. Therefore I did the following changes in order to fix this issue:
- Modified check for total time exceedance taking vacation and transfers into account. This was done by calculating a corrected maximum working time. This new corrected time is the old max working time...
... **minus vacation** (because vacation is part of the legal max working time given by profession).
... **minus predecessor-transfer** (because this is nothing different than daily working time, coming from another month).
... **plus successor-transfer** (because this is the time you do not want to add to this but rather transfer to your next time sheet).
- Added JUnit tests especially for edge cases

For this reason, this PR fixes #39.

### CODE REVIEW

File: src/main/java/checker/MiLoGChecker.java [hunk 1]
```diff
@@ -85,9 +85,18 @@ public CheckerReturn check() throws CheckerException {
...
+        //Legal maximum working time per month
         TimeSpan maxWorkingTime = fullDoc.getProfession().getMaxWorkingTime();
         
-        if (fullDoc.getTotalWorkTime().compareTo(maxWorkingTime) > 0) {
+        //Sum of all daily working times (without pauses!)
+        TimeSpan totalWorkingTime = fullDoc.getTotalWorkTime();
+        
+        //Vacation and transfer corrected time
+        TimeSpan correctedMaxWorkingTime = maxWorkingTime.subtract(fullDoc.getVacation())
+                .subtract(fullDoc.getPredTransfer())
```
Comment (larsk21) [inline src/main/java/checker/MiLoGChecker.java]: Does TimeSpan subtraction work even when the result is negative? I'm unsure what the result of the % operator is in that case. (see /src/main/java/data/TimeSpan.java)

### Scenario:
Given my max working time is 40 hours and I transfer 50 from the month before (maybe transfer 50 to next month as well), then the result of this subtraction would be negative.

Comment (gstuer) [reply src/main/java/checker/MiLoGChecker.java]: Good point. The problem is that this is caused by the TimeSheet constructor itself. I will quickly fix that and commit it. Hold on.

**Comment by gstuer** [comment]
> Are there any tests for the succ transfer alone or is that not necessary?

Good question. I have not wrote these tests yet because I thought the functionality would have been tested enough with the other ones. But I will add these quickly.

File: src/main/java/checker/MiLoGChecker.java (hunk 1 above)
Comment (gstuer) [reply src/main/java/checker/MiLoGChecker.java]: Should be fixed now.

**Comment by gstuer** [comment]
Any more problems existing? Old ones fixed? @larsk21

**Comment by larsk21** [comment]
Looks good to me.
